cmd/root.go

```diff
@@ -47,104 +46,168 @@ var RootCmd = &cobra.Command{
-			for _, diffType := range diffTypes {
-				diff := diffs[diffType]
-				if json {
-					diffResults = append(diffResults, diff.GetStruct())
-				} else {
-					err = diff.OutputText(diffType)
-					if err != nil {
-						glog.Error(err)
-					}
-				}
-			}
+	}()
+
+	diffTypes, err := differs.GetAnalyzers(diffArgs)
+	if err != nil {
+		glog.Error(err.Error())
+		os.Exit(1)
+	}
+	wg.Wait()
+
+	req := differs.DiffRequest{image1, image2, diffTypes}
+	if diffs, err := req.GetDiff(); err == nil {
+		// Outputs diff results in alphabetical order by differ name
+		sortedTypes := []string{}
+		for name := range diffs {
+			sortedTypes = append(sortedTypes, name)
+		}
+		sort.Strings(sortedTypes)
+		glog.Info("Retrieving diffs")
+		diffResults := []utils.DiffResult{}
+		for _, diffType := range sortedTypes {
+			diff := diffs[diffType]
 			if json {
-				err = utils.JSONify(diffResults)
+				diffResults = append(diffResults, diff.GetStruct())
+			} else {
+				err = diff.OutputText(diffType)
 				if err != nil {
 					glog.Error(err)
 				}
 			}
-			fmt.Println()
+		}
+		if json {
+			err = utils.JSONify(diffResults)
+			if err != nil {
+				glog.Error(err)
+			}
+		}
+		fmt.Println()
+		if !save {
 			glog.Info("Removing image file system directories from system")
-			if !save {
-				errMsg := remove(image1.FSPath, true)
-				errMsg += remove(image2.FSPath, true)
-				if errMsg != "" {
-					glog.Error(errMsg)
-				}
+			errMsg := remove(image1.FSPath, true)
+			errMsg += remove(image2.FSPath, true)
+			if errMsg != "" {
+				glog.Error(errMsg)
+			}
+		} else {
+			dir, _ := os.Getwd()
+			glog.Infof("Images were saved at %s as %s and %s", dir, image1.FSPath, image2.FSPath)
+		}
+	} else {
+		glog.Error(err.Error())
+		os.Exit(1)
+	}
+}
+
+func analyzeImage(imageArg string, analyzerArgs []string) {
+	image, err := utils.ImagePrepper{imageArg}.GetImage()
+	if err != nil {
+		glog.Error(err.Error())
+		os.Exit(1)
+	}
+	analyzeTypes, err := differs.GetAnalyzers(analyzerArgs)
+	if err != nil {
+		glog.Error(err.Error())
+		os.Exit(1)
+	}
+
+	req := differs.SingleRequest{image, analyzeTypes}
+	if analyses, err := req.GetAnalysis(); err == nil {
+		// Outputs analysis results in alphabetical order by differ name
+		sortedTypes := []string{}
+		for name := range analyses {
+			sortedTypes = append(sortedTypes, name)
+		}
+		sort.Strings(sortedTypes)
+		glog.Info("Retrieving diffs")
+		analyzeResults := []utils.AnalyzeResult{}
+		for _, analyzeType := range sortedTypes {
+			analysis := analyses[analyzeType]
+			if json {
+				analyzeResults = append(analyzeResults, analysis.GetStruct())
 			} else {
-				dir, _ := os.Getwd()
-				glog.Infof("Images were saved at %s as %s and %s", dir, image1.FSPath, image2.FSPath)
+				err = analysis.OutputText(analyzeType)
+				if err != nil {
+					glog.Error(err)
+				}
+			}
+		}
+		if json {
+			err = utils.JSONify(analyzeResults)
+			if err != nil {
+				glog.Error(err)
+			}
+		}
+		fmt.Println()
```

Comment:
Oh I actually don't even think this is necessary because there's a newline in the template, so I'll just get rid of it...